packages/openapi3-parser/lib/parser.js

```diff
@@ -126,14 +128,35 @@ function parse(source, context) {
     R.unless(isObjectOrAnnotation, createError(context.namespace, 'Source document is not an object')),
     R.unless(isAnnotation, parseOpenAPIObject(context)));
 
-  return R.chain(
+  const parseResult = R.chain(
     R.pipe(
       parseDocument,
       deduplicateUnsupportedAnnotations(context.namespace),
       context.options.generateSourceMap ? filterColumnLine : filterSourceMaps
     ),
     document
   );
+
+  if (!isAnnotation(parseResult.content[0])) {
+    const formatVersion = R.pipe(
```

Comment:
I'd suggest adding a check here for the value of the openapi member element being a string element, and if any of these checks fail I think we should default to the latest OpenAPI 3 version we know about which is `3.0.3`. I think there may be cases where we could get here with an invalid document which cannot be parsed correctly.

Making a test similar to `it('add the format link', () => {` but instead the source is invalid document should verify this, if you set source to be an unexpected value such as `const source = '{}'` or `const source = '[]'` etc.

Reply:
changed here https://github.com/apiaryio/api-elements.js/pull/533/commits/f52cd031afdfb61aa2cf685c37dd758084f49d54

Reply:
check if the value of the OpenAPi member element is a string element https://github.com/apiaryio/api-elements.js/pull/533/commits/88bcb5687ef5bf41922147ee51e1824da6c01830